client/components/data/query-site-comments/index.jsx

```diff
@@ -1,25 +1,66 @@
 /**
  * External dependencies
  */
-import { Component } from 'react';
+import { Component, PropTypes } from 'react';
 import { connect } from 'react-redux';
+import { each, isEmpty, reverse, slice, values } from 'lodash';
 
 /**
  * Internal dependencies
  */
-import { requestCommentsList } from 'state/comments/actions';
+import { requestComment, requestCommentsTree } from 'state/comments/actions';
+import getSiteCommentsTree from 'state/selectors/get-site-comments-tree';
 
-const requestComments = ( { requestSiteComments, siteId, status } ) => (
-	siteId && requestSiteComments( { siteId, status } )
+const requestTree = ( { requestSiteCommentsTree, siteId, status } ) => (
+	siteId && requestSiteCommentsTree( siteId, status )
 );
 
+const requestComments = props => {
+	const {
+		commentsPerPage,
+		commentsTree,
+		page,
+		requestSingleComment,
+		siteId,
+	} = props;
+
+	if ( isEmpty( commentsTree ) ) {
+		return;
+	}
+
+	if ( ! page || ! commentsPerPage ) {
+		each( commentsTree, ( { commentId } ) => {
+			requestSingleComment( siteId, commentId );
+		} );
+		return;
+	}
+
+	const startingIndex = ( page - 1 ) * commentsPerPage;
+	const parsedCommentsTree = reverse( values( commentsTree ) );
+	const commentsPage = slice( parsedCommentsTree, startingIndex, startingIndex + commentsPerPage );
+
+	each( commentsPage, ( { commentId } ) => {
+		requestSingleComment( siteId, commentId );
```

Comment:
@dmsnell suggested to create a query component for this (if we don't have one already) and render it per each comment vs in the list. If we do it that way, this becomes `QuerySiteCommentTree` then